engine/classes/Elgg/BadRequestException.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Elgg;
+
+use Throwable;
+
+/**
+ * Thrown when request is malformatted
+ */
+class BadRequestException extends HttpException {
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function __construct(string $message = "", int $code = 0, Throwable $previous = null) {
+		if (!$message) {
+			$message = elgg_echo('BadRequestException');
```

Comment:
Leave constructors alone and do this in getMessage

Reply:
Can't do because `Exception` marks those as `final`

Reply:
Some use exception factories to clean this up, but this is fine.